src/ray/common/client_connection.cc

```diff
@@ -0,0 +1,110 @@
+#ifndef CLIENT_CONNECTION_CC
+#define CLIENT_CONNECTION_CC
+
+#include "client_connection.h"
+
+#include <boost/bind.hpp>
+
+#include "common.h"
+#include "ray/status.h"
+#include "ray/raylet/format/nm_generated.h"
+
+using namespace std;
```

Comment:
It's weird to see `using namespace std;` but then almost immediately below, we see `std::move`.  My personal preference is to not use this, but up to you.

Reply:
Now that I try to recall, using-directive is banned ([ref](https://google.github.io/styleguide/cppguide.html#Namespaces)).  I prefer if we just adhere to the collective wisdom from dozens (if not hundreds) of production C++ systems.

Reply:
Let's get rid of all `using` statements.

Reply:
We discussed this with @pcmoritz , and the agreement was to disallow `using` in the header files, while allowing it in the implementation`.cc` files, at least for `std`. For most (all?) other namespaces (including `boost`) I am in favor of explicitly specifying them.

Reply:
in fact, I am in favor of explicitly specifying namespaces everywhere (i.e. removing `using namespace`), as long as it's fine with @pcmoritz .

Reply:
We can use things like `using std::shared_ptr;` and shortening namespaces (`namespace asio = boost::asio;`. Perhaps this is an acceptable middle ground.

Reply:
I'm in favor of getting rid of "using namespace" everywhere (I was just arguing for *not* having using namespaces in the headers because it pollutes the namespaces everywhere), but also not using it in source files is even better I think. Abbreviations like elibol mentioned are fine with me.